rcgen/src/certificate.rs

```diff
@@ -567,30 +594,7 @@ impl CertificateParams {
 					// RFC 2986 specifies that attributes are a SET OF Attribute
 					writer.write_set_of(|writer| {
 						if write_extension_request {
-							writer.next().write_sequence(|writer| {
-								let oid =
-									ObjectIdentifier::from_slice(oid::PKCS_9_AT_EXTENSION_REQUEST);
-								writer.next().write_oid(&oid);
-								writer.next().write_set(|writer| {
-									writer.next().write_sequence(|writer| {
-										// Write key_usage
-										self.write_key_usage(writer.next());
-										// Write subject_alt_names
-										self.write_subject_alt_names(writer.next());
-										self.write_extended_key_usage(writer.next());
-
-										// Write custom extensions
-										for ext in custom_extensions {
-											write_x509_extension(
-												writer.next(),
-												&ext.oid,
-												ext.critical,
-												|writer| writer.write_der(ext.content()),
-											);
-										}
-									});
-								});
-							});
+							self.write_extension_request_attribute(writer.next());
```

Comment:
Nit: suggest moving the `write_extension_request` check into the new method.

Reply:
I considered doing this, but it makes `subject_alt_names`, `key_usages`, and `custom_extensions` unused, which we deny here:

https://github.com/rustls/rcgen/blob/02d5c6096ec54b491f60581f7e51e83bdd764d84/rcgen/src/certificate.rs#L507-L522

The alternative is making `CertificateParams::write_extension_request_attribute` static, i.e. not a function of `&self`. I'm fine with the way it's currently written, so this one is your call!

Reply:
Okay, I guess this is a reasonable mechanism for making sure we're taking everything into account.